src/main/java/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,60 @@
+package bookstore.repository.impl;
+
+import bookstore.entity.Book;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        EntityManager entityManager = null;
+        try {
+            entityManager = entityManagerFactory.createEntityManager();
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new EntityNotFoundException("Can't insert book into db " + book, e);
```

Comment:
you're trying to save an entity but at the same time throw an exception, that entity not found


weird, don't you think?)

Reply:
isn't it edge case? I'll remove it btw